### PR TITLE
Feature

### PR DESCRIPTION
**Pull request by NikolayStrepetov**

### CODE REVIEW

File: sources/example.cpp [hunk 1]
```diff
@@ -1,13 +1,190 @@
...
-{
-    std::string tmp_string = "Hello from ";
-    tmp_string.append(__FUNCTION__);
+BowlingCalculator::BowlingCalculator() : currentRoll(0) {
+    std::fill_n(rolls, 21, 0);
+}
+
+void BowlingCalculator::roll(int pins) {
+    if (isGameOver()) {
+        throw std::runtime_error("The Game is already over!");
```
Comment (antonKirpich) [inline sources/example.cpp:14]: If a function can generate an exception, it is also worth to cover these edge cases by unit tests.

File: includes/example.h [hunk 2]
```diff
@@ -1,4 +1,21 @@
...
+    void roll(int pins);
+    int totalScore() const;
+    int getCurrentFrame() const;
+    int getFrameScore(int frame) const;
+    std::vector<int> getRollsInFrame(int frame) const;
+    bool isGameOver() const;
 
-std::string dummy_function();
+private:
+    int rolls[21];
```
Comment (antonKirpich) [inline includes/example.h:14]: Good practice is not to use magic numbers. It is better to use constants or define expressions.
For example:
```
#define MAX_ROLLS_PER_GAME   21
// ....
int rolls[MAX_ROLLS_PER_GAME];
```

File: sources/example.cpp [hunk 3]
```diff
@@ -1,13 +1,190 @@
...
+        frame ++;
+    }
+
+    return frame;
+}
+
+int BowlingCalculator::getFrameScore(int frame) const {
+    if (frame < 1 || frame > 10) {
+        throw std::invalid_argument("Frame must be between 1 and 10");
+    }
```
Comment (antonKirpich) [inline sources/example.cpp:63]: Same suggestion to cover generated exceptions

File: sources/example.cpp [hunk 4]
```diff
@@ -1,13 +1,190 @@
...
+        frameScore = rolls[rollIndex] + rolls[rollIndex + 1];
+    }
+
+    return frameScore;
+}
+
+std::vector<int> BowlingCalculator::getRollsInFrame(int frame) const {
+    if (frame < 1 || frame > 10) {
+        throw std::invalid_argument("Frame must be between 1 and 10");
+    }
```
Comment (antonKirpich) [inline sources/example.cpp:95]: Same suggestion to cover generated exceptions

File: sources/example.cpp [hunk 5]
```diff
@@ -1,13 +1,190 @@
...
+        else {
+            rollIndex += 2;
+        }
+        frame ++;
+    }
+
+    return frame;
+}
+
+int BowlingCalculator::getFrameScore(int frame) const {
```
Comment (antonKirpich) [inline sources/example.cpp:60]: `getFrameScore` function always returns a value, even if the frame is not complete. Frame scores should only be counted after the frame and bonus throws completed.
Suggestion: return -1 or throw an exception if the frame is not complete.

File: sources/example.cpp [hunk 6]
```diff
@@ -1,13 +1,190 @@
...
+
+    while (frameIndex < frame -1) {
+        if (isStrike(rollIndex)) {
+            rollIndex ++;
+        }
+        else {
+            rollIndex += 2;
+        }
+        frameIndex ++;
+    }
```
Comment (antonKirpich) [inline sources/example.cpp:77]: The same code is also present in `getRollsInFrame` function, we can move it to a separate helper function.
